signer/cosigner_nonce_cache.go

```diff
@@ -116,16 +134,23 @@ func NewCosignerNonceCache(
 	getNoncesInterval time.Duration,
 	getNoncesTimeout time.Duration,
 	threshold uint8,
+	pruner NonceCachePruner,
 ) *CosignerNonceCache {
-	return &CosignerNonceCache{
+	cnc := &CosignerNonceCache{
 		logger:            logger,
 		cache:             NewNonceCache(),
 		cosigners:         cosigners,
 		leader:            leader,
 		getNoncesInterval: getNoncesInterval,
 		getNoncesTimeout:  getNoncesTimeout,
 		threshold:         threshold,
+		pruner:            pruner,
 	}
+	if pruner == nil {
```

Comment:
Add a comment that this is expected to be nil for now, but the unit test sets it with a mockPruner.